autora/cycle/plot_utils.py

```diff
@@ -472,3 +481,102 @@ def plot_results_panel_3d(
     )
 
     return fig
+
+
+def cycle_default_score(cycle: Cycle, x_vals: np.ndarray, y_true: np.ndarray):
+    """
+    Calculates score for each cycle using the estimator's default scorer.
+    Args:
+        cycle: AER Cycle object that has been run
+        x_vals: Test dataset independent values
+        y_true: Test dataset dependent values
+
+    Returns:
+        List of scores by cycle
+    """
+    l_scores = [s.score(x_vals, y_true) for s in cycle.data.theories]
+    return l_scores
+
+
+def cycle_specified_score(
+    scorer: Callable, cycle: Cycle, x_vals: np.ndarray, y_true: np.ndarray, **kwargs
+):
+    """
+    Calculates score for each cycle using specified sklearn scoring function.
+    Args:
+        scorer: sklearn scoring function
+        cycle: AER Cycle object that has been run
+        x_vals: Test dataset independent values
+        y_true: Test dataset dependent values
+        **kwargs: Keyword arguments to send to scoring function
+
+    Returns:
+
+    """
+    # Get predictions
+    if "y_pred" in inspect.signature(scorer).parameters.keys():
+        d_y_pred = _theory_predict(cycle, x_vals, predict_proba=False)
+    elif "y_score" in inspect.signature(scorer).parameters.keys():
+        d_y_pred = _theory_predict(cycle, x_vals, predict_proba=True)
+
+    # Score each cycle
+    l_scores = []
+    for i, y_pred in d_y_pred.items():
+        l_scores.append(scorer(y_true, y_pred, **kwargs))
+
+    return l_scores
+
+
+def plot_cycle_score(
+    cycle: Cycle,
+    X: np.ndarray,
+    y_true: np.ndarray,
+    scorer: Optional[Callable] = None,
+    x_label: str = "Cycle",
+    y_label: Optional[str] = None,
+    figsize: Tuple[float, float] = rcParams["figure.figsize"],
+    scorer_kw: dict = {},
+    plot_kw: dict = {},
+) -> plt.Figure:
+    """
+    Plots scoring metrics of cycle's theories given test data.
+    Args:
```

Comment:
I am wondering a bit about the y-limits. Here, it allows it to move freely based on the data, but this can be quite misleading when comparing different plots. For example, in example/cycle_scoring.ipynb you see some plots start at ~75% on cycle 1 and others at ~92% but because the y-limit is free to move, they look like similar progressions across cycles (despite the former gaining much more accuracy than the latter). 

I am wondering if you might find it useful to add some defaults (e.g., lower = 50%, higher = 100%) that can be modified by the user if necessary. Or even the lower bound can have thresholds so that it chooses 0%, 25%, or 50% depending where the lower bound of the data lies.

Reply:
Yeah that's a good idea. I think we can add a keyword for `ylim` and `xlim` so the user can specify if they want to zoom on a specific part of the plot.

Reply:
Ok, I've added this feature. The last cell of the notebook example has been updated with an example of usage.

Reply:
Awesome, looks good!